### PR TITLE
[ifx](routine-load) fix show routine load task error

### LINKED ISSUE #8194: [Bug] [routine load] the tnxStatus of "show routine load task" result are not correct
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

`TxnStatus` of the result of following stmt  is always `UNKNOWN`

`show routine load task where JobName = "test_routine"; `

```
+-----------------------------------+-------+-----------+
| TaskId                            | TxnId | TxnStatus | 
+-----------------------------------+-------+-----------+
| 6f8566ac62674a7b-b2e103fcc5521c9a | 18    | UNKNOWN   |
+-----------------------------------+-------+-----------+
```

### What You Expected?

`TxnStatus` of the result of following stmt should be the same as shown in `show proc '/transactions/DbId/running/'`

for example:

```
+-----------------------------------+-------+-----------+
| TaskId                            | TxnId | TxnStatus | 
+-----------------------------------+-------+-----------+
| 6f8566ac62674a7b-b2e103fcc5521c9a | 18    | PREPARE   |
+-----------------------------------+-------+-----------+
```

### How to Reproduce?

show routine load task where JobName = "JOB_NAME";

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Henry2SS**
# Proposed changes

fix show routine load task's status error.
tested locally.
Issue Number: close #8194 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (No Need)
3. Has document been added or modified: (No Need)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java [hunk 1]
```diff
@@ -1299,7 +1300,14 @@ protected abstract boolean checkCommitInfo(RLTaskTxnCommitAttachment rlTaskTxnCo
 
     public List<List<String>> getTasksShowInfo() {
         List<List<String>> rows = Lists.newArrayList();
-        routineLoadTaskInfoList.stream().forEach(entity -> rows.add(entity.getTaskShowInfo()));
+        routineLoadTaskInfoList.stream().forEach(entity -> {
+            try {
+                entity.setTxnStatus(Catalog.getCurrentCatalog().getGlobalTransactionMgr().getDatabaseTransactionMgr(dbId).getTransactionState(entity.getTxnId()).getTransactionStatus());
+                rows.add(entity.getTaskShowInfo());
+            } catch (AnalysisException e) {
+                LOG.warn("failed to setTxnStatus db: {}, txnId: {}", dbId, entity.getTxnId());
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java:1308]:
```suggestion
                LOG.warn("failed to setTxnStatus db: {}, txnId: {}, err: ", dbId, entity.getTxnId(), e.getErrorMsg());
```

Comment (Henry2SS) [reply fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java]: Done.

**Review by yangzhg — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java [hunk 2]
```diff
@@ -1299,7 +1300,14 @@ protected abstract boolean checkCommitInfo(RLTaskTxnCommitAttachment rlTaskTxnCo
 
     public List<List<String>> getTasksShowInfo() {
         List<List<String>> rows = Lists.newArrayList();
-        routineLoadTaskInfoList.stream().forEach(entity -> rows.add(entity.getTaskShowInfo()));
+        routineLoadTaskInfoList.stream().forEach(entity -> {
+            try {
+                entity.setTxnStatus(Catalog.getCurrentCatalog().getGlobalTransactionMgr().getDatabaseTransactionMgr(dbId).getTransactionState(entity.getTxnId()).getTransactionStatus());
+                rows.add(entity.getTaskShowInfo());
+            } catch (AnalysisException e) {
+                LOG.warn("failed to setTxnStatus db: {}, txnId: {}, err: {}", dbId, entity.getTxnId(), e.getErrorMsg());
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java:1308]:
```suggestion
                LOG.warn("failed to setTxnStatus db: {}, txnId: {}, err: {}", dbId, entity.getTxnId(), e.getMessage());
```

**Review by morningman — CHANGES_REQUESTED** [review]
Please fix compile error

File: fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java (hunk 2 above)
Comment (Henry2SS) [reply fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java]: Done

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**